Runtime/Scripts/MobiledgeXUDPClient.cs

```diff
@@ -0,0 +1,144 @@
+    public class MobiledgeXUDPClient : IDisposable
+    {
+        private UdpClient udpClient;
+        private string host;
+        private int port;
+        Thread receiveThread { get; set; }
+        Thread sendThread { get; set; }
+        static UTF8Encoding encoder;
+        public ConcurrentQueue<byte[]> receiveQueue { get; }
+        public BlockingCollection<ArraySegment<byte>> sendQueue { get; }
+        public bool run = true;
+        const int MAXPAYLOADSIZE = 508; // max payload size guaranteed to be deliverable (not guaranteed to be delivered)
+
+        public MobiledgeXUDPClient(string hostName, int sendPort, int receivePort)
+        {
+            try
+            {
+                udpClient = new UdpClient(receivePort);
+                udpClient.DontFragment = true;
+
+            }
+            catch (Exception e)
+            {
+                Debug.LogError("Failed to listen for UDP at port " + receivePort + ": " + e.Message);
+                return;
+            }
+
+            host = Dns.GetHostAddresses(hostName)
+                .First(ip => ip.AddressFamily == AddressFamily.InterNetwork
+                || ip.AddressFamily == AddressFamily.InterNetworkV6).ToString();
+            port = sendPort;
+            encoder = new UTF8Encoding();
+            receiveQueue = new ConcurrentQueue<byte[]>();
+            receiveThread = new Thread(RunReceive);
+            receiveThread.Start();
+            sendQueue = new BlockingCollection<ArraySegment<byte>>();
+            sendThread = new Thread(RunSend);
+            sendThread.Start();
+            Connect();
+        }
+
+        public void Connect()
+        {
+            run = true;
+        }
+
+        public void Send(string message)
+        {
+
+            byte[] buffer = encoder.GetBytes(message);
+            if (buffer.Length > MAXPAYLOADSIZE)
+            {
+                Debug.LogError("Max UDP payload size is 508 bytes, trying slicing your message to suit the max payload size");
+                return;
+            }
+            var sendBuf = new ArraySegment<byte>(buffer);
+            sendQueue.Add(sendBuf);
+        }
+
+        public void Send(byte[] buffer)
+        {
+            if (buffer.Length > MAXPAYLOADSIZE)
+            {
+                Debug.LogError("Max UDP payload size is 508 bytes, trying slicing your buffer to suit the max payload size");
+                return;
+            }
+            var sendBuf = new ArraySegment<byte>(buffer);
+            sendQueue.Add(sendBuf);
+        }
+
+        public async void RunSend()
+        {
+            ArraySegment<byte> msg;
+            Debug.Log("RunSend entered.");
+            while (run)
+            {
+                while (!sendQueue.IsCompleted)
+                {
+                    msg = sendQueue.Take();
+                    long count = sendQueue.Count;
+                    Debug.Log("Dequeued this message to send: " + msg + ", queueSize: " + count);
```

Comment:
Config for verbose mode?

Reply:
Thats a good suggestion we should do that for the whole SDK , will open a ticket for that